extensions/panache/hibernate-orm-panache-common/runtime/src/main/java/io/quarkus/hibernate/orm/panache/Panache.java

```diff
@@ -0,0 +1,64 @@
+package io.quarkus.hibernate.orm.panache;
+
+import java.util.Collections;
+import java.util.Map;
+
+import javax.persistence.EntityManager;
+import javax.transaction.TransactionManager;
+
+import io.quarkus.arc.Arc;
+import io.quarkus.hibernate.orm.PersistenceUnit;
+import io.quarkus.hibernate.orm.runtime.PersistenceUnitUtil;
+
+/**
+ * Utility class for Panache.
+ *
+ * @author Stéphane Épardaud
+ */
+public class Panache {
```

Comment:
ah.  i wasn't aware that was intended for public consumption.  i'll see what's drifted and restore it.

Reply:
The problem with these methods is that they treat `JpaOperations` as a bag of static methods but it's not any more.  So those methods would have to either change to take in the the operations reference, which would be a weird API, or just go away.  Are users actually actively using these methods?  Because I'm not seeing a non-breaking way for them to remain with everything else as it is.

Reply:
Well, yes, those methods are used in the wild, that's the public API. I'm not sure what you mean, we currently already have several versions of this class (different packages), so as long as we keep them we know what instance of operations they delegate to, no?

Reply:
right.  they would need to have a reference to an operations instance.  not sure how to get that in there without changing the API.  but then, we don't *want* them using the operations classes directly because those are *not* public.

Reply:
Just make them forward to the right `INSTANCE`, I don't see the problem.

Reply:
well, there are two issues is see off hand (and @loicmathieu can probably weigh in here):

1. That's still a breaking change to a public API.  I personally don't see the value in this public API but that might be a discussion for another time.
1. We've been mostly working on the assumption that the operations implementation classes are internal implementation details and discouraging use outside of the generated delegations from entity/repository types.  this would officially commit us to a public, stable API.  That's not necessarily a bad thing as that API is unlikely to radically change but I do think it's a decision we should consciously make.

Adding that as a param is by far the simplest thing to do I just want to make sure we're actively choosing the consequences of it.

Reply:
Definitely not as a param:

```java
public class Panache {
  public static void setRollbackOnly() {
    JpaOperations.INSTANCE.setRollbackOnly();
 }
}
```

And do the same for the other existing `Panache` classes and other methods, and we're done.

Reply:
That would create a hard dep on `hibernate-orm-panache` even for kotlin apps which could be awkward/confusing.  fwiw, the reactive form of `Panache` is a much smaller API which suggest some asymmetry in requirements/design.  Do users *really* need another helper class like this?

Reply:
I'm not sure what you mean about this. Currently every Panache extension has that `Panache` class. You're trying to remove it. I think you can't because people are using it. I'm not proposing to add anything new.

Reply:
JpaOperations.INSTANCE lives in `hibernate-orm-panache` which `quarkus-hibernate-orm-panache-kotlin` does not depend on.  In order for `Panache` to function, those kotlin apps would have to import the java implementation as well.  the kotlin module does not define its `Panache` so it can't be bound to a different reference.  Now we can push down the `Panache` definitions to the language specific modules but it'd be one more point of duplication.  I have plans in the works to monitor all such duplications for divergence so maybe that's not a big deal.  But, again, it should be a conscious choice we all make.

Reply:
If the Kotlin version doesn't have this interface ATM, then it differs from the Java one, just move it back to the ORM module where it started. The Mongo ones have it, under their own package, so no problem there.

Reply:
I can definitely push these types down to the specific modules.  Still trying to track down why switching from the envers dep to hibernate-orm results in that arc/injection failure.

Reply:
OK.  Simpler solution:  I can simply make those methods static on `AbstractJpaOperations` since they don't need any specialization of types between java/kotlin, e.g.  I have tests running locally now and will push as soon as they pass here.

Reply:
everything seems to be passing now.  I've done a little bit of pom clean up where we had redundant deps.  it helped me to better track where the changes needed to be made to resolve some of the envers bits.